File: common/src/main/java/com/wdz/common/MyApplication.java
package com.wdz.common;

import android.app.Application;

import androidx.lifecycle.MutableLiveData;

import com.alibaba.android.arouter.launcher.ARouter;
import com.tencent.mmkv.MMKV;
import com.wdz.common.net.response.LoginResponse;
import com.wdz.common.util.Log;
import com.wdz.common.util.MmkvUtils;

/**
 * @Author dezhi.wang
 * @Date 2021/1/13 15:39
 */
public class MyApplication extends Application {
    private final String TAG = this.getClass().getSimpleName();
    private MyApplication instance;
    private LoginResponse loginResponse;
    private MutableLiveData<LoginResponse> loginResponseMutableLiveData = new MutableLiveData<>();
    @Override
    public void onCreate() {
        super.onCreate();
        instance = this;
    }

    public MyApplication getInstance(){
        return instance;
    }
    public void setUserInfo(LoginResponse loginResponse){
        this.loginResponse = loginResponse;
        loginResponseMutableLiveData.postValue(loginResponse);
    }
    public MutableLiveData<LoginResponse> getUerInfo(){
        //如果loginResponse为null，重新从mmkv读取
        if (loginResponse == null){
            LoginResponse loginUser = MmkvUtils.getLoginUser();
            setUserInfo(loginUser);
        }
        return loginResponseMutableLiveData;

    }
}

File: kt_common/src/main/java/com/wdz/ktcommon/MyApplication.kt
package com.wdz.ktcommon

import android.app.Application
import android.content.Context
import androidx.lifecycle.MutableLiveData

import com.wdz.ktcommon.response.LoginResponse
import com.wdz.ktcommon.utils.getLoginUser
import kotlin.properties.Delegates

/**

 * @Author dezhi.wang

 * @Date 2021/3/31 10:13

 */
class MyApplication: Application() {
    private var loginResponse: LoginResponse? = null
    private val loginResponseMutableLiveData =
        MutableLiveData<LoginResponse>()
    companion object{
        // TODO: 2021/3/31 延迟加载 lateinit->引用类型   Delegates.notNull->基本数据类型，引用类型（延迟初始化的委托类）
        var context:Context by Delegates.notNull()

    }
    override fun onCreate() {
        super.onCreate()
        context = applicationContext
    }

    fun setUserInfo(loginResponse: LoginResponse) {
        this.loginResponse = loginResponse
        loginResponseMutableLiveData.postValue(loginResponse)
    }

    fun getUerInfo(): MutableLiveData<LoginResponse> {
        //如果loginResponse为null，重新从mmkv读取
        if (loginResponse == null) {
            val loginUser = getLoginUser()
            if (loginUser != null) {
                setUserInfo(loginUser)
            }
        }
        return loginResponseMutableLiveData
    }
}